golem-worker-executor-base/src/services/worker_event.rs

```diff
@@ -173,20 +97,26 @@ impl WorkerEventService for WorkerEventServiceDefault {
 
             let _ = self.sender.send(event.clone());
         }
-        let _ = unsafe { Producer::new(&self.ring) }.push(event);
+
+        let mut ring_prod = self.ring_prod.lock().unwrap();
+        while ring_prod.try_push(event.clone()).is_err() {
```

Comment:
does this mean we keep dropping events until we have a free slot in the buffer?
is that something that should be logged?
is it possible that this will need to be run more then once? (as in is this while just for simplicity, or is there an actual possibility for looping?  because of the lock i guess not)